runtime/src/bank.rs

```diff
@@ -1845,29 +1845,25 @@ impl Bank {
         &self,
         signature: &Signature,
     ) -> Option<Result<()>> {
-        if let Some(status) = self.get_signature_confirmation_status(signature) {
-            if status.slot == self.slot() {
-                return Some(status.status);
+        if let Some((slot, status)) = self.get_signature_status_slot(signature) {
+            if slot <= self.slot() {
```

Comment:
@jstarry , this makes the changes you made in #9019 slightly less strict; do you think it is acceptable?
Can you think of any cases are there where this could result in multiple notifications?

This is the reason that `test_rpc_subscriptions` was failing with higher transaction counts (the notification loop is slightly slower with this pr, so it was more obvious). Here's why: if the notification loop can't process all the subscriptions that should be notified in a slot before the next slot boundary, they never find a slot that meets both the confirmations match in `check_confirmations_and_notify` and this case here, because the selected bank is now the bank for the next slot.

Reply:
Ah makes sense. This is fine due to the auto unsubscribe behavior for signature subscriptions. I'll fix the broken test and will see about speeding up the notification sender.

Reply:
Oh there is an issue actually. The slot context will be incorrect with this change. If you subscribe for a signature that has received X confirmations and receive a notification with context of slot: Y then you would assume that Y has X confirmations

Reply:
Eh, actually I don't think it's an issue as long as we make it clear that a notification will be sent if the signature has at least X confirmations. In that case, you wouldn't make any assumptions about the context slot's confirmations. What do you think?

Reply:
See here for more details: https://github.com/solana-labs/solana/pull/9138/commits/a4eb8c2b70a148021253e9f393336177ec3c00a0#diff-153526d7b6a2099938f98c6cfd2e98a9R926-R936

Reply:
Yeah, I don't think we can guarantee that a transaction will be rooted at exactly slot + (MAX_LOCKOUT - confirmations). But it should be rooted BY that slot.

Reply:
Oh, I missed the part where you said you were going to fix that test. Thanks, @jstarry !

Reply:
@CriesofCarrots I ended up fixing the issue in the subscriptions test! You could switch this back to `slot == self.slot()` but I think it's fine to keep it as is! We could have "exact" match to be an subscription param if we want. Here's the link to the PR: https://github.com/solana-labs/solana/pull/9142